### PR TITLE
No worker machinepool HealthCheck

### PR DESCRIPTION
**Pull request by a7vicky**
### What type of PR is this?
_(feature)_

### What this PR does / why we need it?
- Send notification during upgrade in case MUO is unable to perform compute capacity reservation because the default worker machine pool is deleted 
- As part of the health check it verify the default worker machine pool exists in the cluster or not and returns the value accordingly
- Set metrics if the health check fails

### Which Jira/Github issue(s) this PR fixes?
_Fixes #_[OSD-22459](https://issues.redhat.com/browse/OSD-22459)

### Pre-checks (if applicable):
- [x] Tested latest changes against a cluster
- [ ] Included documentation changes with PR

### CODE REVIEW

**Comment by shitaljante** [comment]
lgtm

File: pkg/upgraders/healthcheckstep.go [hunk 1]
```diff
@@ -40,6 +41,18 @@ func (c *clusterUpgrader) PreUpgradeHealthCheck(ctx context.Context, logger logr
 		}
 		return false, err
 	}
+
+	if c.upgradeConfig.Spec.CapacityReservation {
+		ok, err := c.scaler.CanScale(c.client, logger)
+		if !ok {
+			c.metrics.UpdateMetricHealthcheckFailed(c.upgradeConfig.Name, metrics.DefaultWorkerMachinepoolNotFound)
```
Comment (ravitri) [inline pkg/upgraders/healthcheckstep.go:48]: So currently, there's no notification (`c.notifier.Notify(notifier.MuoStateHealthCheck)`) sent for this failure in pre-healthcheck. Is that missed?

**Comment by ravitri** [comment]
/lgtm

Thanks a lot Abhishek!

**Pull request merged**